cmd/ipfs/Rules.mk

```diff
@@ -13,7 +13,20 @@ PATH := $(realpath $(d)):$(PATH)
 # DEPS_OO_$(d) += merkledag/pb/merkledag.pb.go namesys/pb/namesys.pb.go
 # DEPS_OO_$(d) += pin/internal/pb/header.pb.go unixfs/pb/unixfs.pb.go
 
-$(d)_flags =-ldflags="-X "github.com/ipfs/go-ipfs".CurrentCommit=$(git-hash)"
+DATE := $(shell date +%FT%T%z)
+USER := $(shell whoami)
+GIT_VERSION := $(shell git --no-pager describe --tags --always --dirty)
```

Comment:
Maybe we should just set this in an init function in `version.go`. That way, we can use the official version.